src/Microsoft.Data/DataFrame.cs

```diff
@@ -34,6 +35,150 @@ public DataFrame(IList<BaseColumn> columns)
+                        break;
+                    case ArrowTypeId.UInt32:
+                        PrimitiveArray<uint> arrowUintArray = (PrimitiveArray<uint>)arrowArray;
+                        ReadOnlyMemory<byte> uintValueBuffer = arrowUintArray.ValueBuffer.Memory;
+                        ReadOnlyMemory<byte> uintNullBitMapBuffer = arrowUintArray.NullBitmapBuffer.Memory;
+                        dataFrameColumn = new PrimitiveColumn<uint>(field.Name, uintValueBuffer, uintNullBitMapBuffer, arrowArray.Length, arrowArray.NullCount);
+                        break;
+                    case ArrowTypeId.UInt64:
+                        PrimitiveArray<ulong> arrowUlongArray = (PrimitiveArray<ulong>)arrowArray;
+                        ReadOnlyMemory<byte> ulongValueBuffer = arrowUlongArray.ValueBuffer.Memory;
+                        ReadOnlyMemory<byte> ulongNullBitMapBuffer = arrowUlongArray.NullBitmapBuffer.Memory;
+                        dataFrameColumn = new PrimitiveColumn<ulong>(field.Name, ulongValueBuffer, ulongNullBitMapBuffer, arrowArray.Length, arrowArray.NullCount);
+                        break;
+                    case ArrowTypeId.Decimal:
+                    case ArrowTypeId.Binary:
+                    case ArrowTypeId.Date32:
+                    case ArrowTypeId.Date64:
+                    case ArrowTypeId.Dictionary:
+                    case ArrowTypeId.FixedSizedBinary:
+                    case ArrowTypeId.HalfFloat:
+                    case ArrowTypeId.Interval:
+                    case ArrowTypeId.List:
+                    case ArrowTypeId.Map:
+                    case ArrowTypeId.Null:
+                    case ArrowTypeId.Struct:
+                    case ArrowTypeId.Time32:
+                    case ArrowTypeId.Time64:
+                    default:
+                        throw new NotImplementedException(nameof(fieldType.Name));
+                }
+                _table.InsertColumn(ColumnCount, dataFrameColumn);
+                fieldIndex++;
+            }
+        }
+
+        public IEnumerable<RecordBatch> AsArrowRecordBatches()
+        {
+            Apache.Arrow.Schema.Builder schemaBuilder = new Apache.Arrow.Schema.Builder();
+
+            int columnCount = ColumnCount;
+            for (int i = 0; i < columnCount; i++)
+            {
+                BaseColumn column = Column(i);
+                Field field = column.Field();
+                schemaBuilder.Field(field);
+            }
+
+            Schema schema = schemaBuilder.Build();
+            List<Apache.Arrow.Array> arrays = new List<Apache.Arrow.Array>();
+
+            int recordBatchLength = Int32.MaxValue;
+            int numberOfRowsInThisRecordBatch = (int)Math.Min(recordBatchLength, RowCount);
+            long numberOfRowsProcessed = 0;
+
+            // Sometime Spark passes in an empty DataFrame 
```

Comment:
I think this comment could use some work. It isn't understandable in its current form to someone new in the code.